tools/tpm2_createpolicy.c

```diff
@@ -0,0 +1,333 @@
+    nonceCaller.t.size = 0;
+    
+    // Start policy session.
+    TPM_RC rval = tpm_session_start_auth_with_params(ctx->sapi_context,
+        &ctx->policy_session, TPM_RH_NULL, 0, TPM_RH_NULL, 0, &nonceCaller, 
+        &encryptedSalt, TPM_SE_TRIAL, &symmetric, TPM_ALG_SHA256);
+
+    if (rval != TPM_RC_SUCCESS) {
+        LOG_ERR("Failed tpm session start auth with params\n");
+        return rval;        
+    }
+
+    // Send policy command.
+    rval = ( *build_policy_function )( ctx);
+    if (rval!=TPM_RC_SUCCESS) {
+        LOG_ERR("Failed build_policy_function\n");
+        return rval;
+    }
+
+    // Get policy hash.
+    ctx->policy_digest.t.size = 0;
+    INIT_SIMPLE_TPM2B_SIZE( ctx->policy_digest );
+    rval = Tss2_Sys_PolicyGetDigest( ctx->sapi_context, 
+        ctx->policy_session->sessionHandle, 0, &ctx->policy_digest, 0 );
+    if (rval!=TPM_RC_SUCCESS) {
+        LOG_ERR("Failed Policy Get Digest\n");
+        return rval;
+    }
+
+    printf("Created Policy Digest:\n");
+    string_bytes_print_tpm2b(&ctx->policy_digest.b);
+
+    //save the policy buffer in a file for use later
+    if (saveDataToFile(ctx->policyfile, (UINT8 *)&ctx->policy_digest.t.buffer,
+        ctx->policy_digest.t.size)) {
+        LOG_ERR("Failed to save policy digest into file \"%s\"", ctx->policyfile);
+        return -1;
+    }
+
+    // Need to flush the session here.
+    rval = Tss2_Sys_FlushContext( ctx->sapi_context,
+        ctx->policy_session->sessionHandle );
+    if( rval!=TPM_RC_SUCCESS ) {
+        LOG_ERR("Failed Flush Context\n");
+        return rval;
+    }
+
+    // And remove the session from sessions table.
+    rval = tpm_session_auth_end(ctx->policy_session);
+
+    return rval;
+}
+
+bool parse_policy_type(createpolicypcr_ctx *ctx, char *argv[]) {
+
+    if(ctx->policy_type_pcr_flag) {
+        if (!ctx->policy_file_flag || !ctx->pcr_index_flag) {
+            showArgMismatch(argv[0]);
+            return false;
+        } else {
+            LOG_INFO("Policy File = %s\n", ctx->policyfile);
+            LOG_INFO("PCR Index= %d\n",ctx->pcr_index);
+            UINT32 rval = build_policy(ctx, build_pcr_policy);
+            if (rval != TPM_RC_SUCCESS) {
+                LOG_ERR("Failed build_policy\n");
+                return false;
+            }
+        }
+    }
+    return true;
+}
+
+static bool init(int argc, char *argv[], char *envp[], createpolicypcr_ctx *ctx) {
+    
+    struct option sOpts[] = {
+            { "policy-file",     required_argument,  NULL,  'f' },
+            { "policy-PCR",      no_argument,        NULL,  'P' },
+            { "pcr-index",       required_argument,  NULL,  'i' },
+            { "pcr-alg",         required_argument,  NULL,  'g' },
+            { "pcr-input-file",  required_argument,  NULL,  'F' },
+            { NULL,              no_argument,        NULL, '\0' },
+    };
+
+    if (argc == 1) {
+        execute_man(argv[0], envp);
+        return false;
+    }
```

Comment:
We're starting to build inconsitent behavior between the tools, remove this execute_man() and just let it fall into showArgMismatch() and then we can mark envp off of execute_tool as unused.